src/proteomes/adapters/proteomesConverter.ts

```diff
@@ -78,12 +78,12 @@ export type ProteomesAPIModel = {
   taxonomy: Taxonomy;
   modified: string;
   proteomeType: ProteomeType;
-  components: Component[];
+  components?: Component[];
```

Comment:
Do you know these are going to be optional from the API?

Reply:
All of the fields that correspond to an API `field` value should be optional TBH, as to not rely on them in other renderers.
Or maybe the ResultPage should operate on `Partial<>` versions of the payload to be on the safe side?

Reply:
So far in the code base we are assuming the API will provide objects with some guaranteed attributes. I think assuming nothing is guaranteed is a bigger question that we would want to discuss with the team.

Reply:
Not for UniProtKB, but indeed we do for other namespaces. I reverted that for now and we should take a more general approach.
I had started added these `?:` a bit randomly when I got an error popping on my screen after toggling some columns.
I have narrowed that down to when I was switching the `lineage` column so I changed the `.map` to `?.map` instead. It's a race condition when the column is rendered but the new data with the `lineage` field isn't there yet.

A bit worried that might happen a bit randomly depending on which columns get added or removed.

Reply:
I haven't seen that race condition before now but now I do. Seems the problem would lie in column customisation & results view?

Reply:
Yes, I guess so. There's some dark magic in the ResultsView component to make this kind of thing work, and it's hard to test automatically, but I guess we're suppose to tackle this race condition there, in a similar way we're doing for the race condition when switching from table to card.